Automattic/woocommerce-payments
# Pass Blocks checkout appearance on init WooPay

#### Changes proposed in this Pull Request
This PR is part of [Spike – Global Themes Support](https://woopayp2.wordpress.com/2024/07/01/spike-global-themes-support/) [proof of concept](https://woopayp2.wordpress.com/2024/07/01/spike-global-themes-support/#proof-of-concept-for-option-3-preferred).

We reuse `getAppearance` to select styles by elements from the blocks checkout page and send them to WooPay on initialization.

EPIC https://github.com/Automattic/woopay/issues/2781
SPIKE [here](https://woopayp2.wordpress.com/2024/07/01/spike-global-themes-support/#proof-of-concept-for-option-3-preferred)
PT [here](https://woopayp2.wordpress.com/2024/07/11/project-thread-woopay-global-themes-support/)


https://github.com/user-attachments/assets/0f50d4fb-1745-4ef4-82d2-368ab37b8882



#### Testing instructions
* Test with [WooPay PR](https://github.com/Automattic/woopay/pull/2805)
* Go to WooPay admin -> WCPay Dev and enable `WooPay global theme support` feature flag
* Update the merchant styles:
1. Install and activate a theme that supports the site editor from admin -> appearance -> theme
2. Update typography by visiting Theme -> Editor -> Styles
OR
1. Update merchant styles ( ex. background color, text color, border styles, etc ) without using [theme that supports the site editor](https://wordpress.com/themes/filter/full-site-editing) from wp-admin appearance

Test 1:
1. Test this PR with the latest WooPayments version that include `getAppearance`
1. As an existing WooPay shopper, go to the blocks checkout page and initialize WooPay with the **email field redirect**
2. Should see WooPay support background color, color, and font family
3. Test `Return to cart` -> Blocks checkout page -> WooPay button
4. Should see WooPay support background color, color, and font family
5. Test `Checkout as guest` and initialize WooPay as a new WooPay shopper with the **email field redirect**
6. Should see WooPay support background color, color, and font family
7. Test `Return to cart` -> Blocks checkout page
8. As a new WooPay shopper, initialize WooPay with the WooPay button
9. Should see WooPay support background color, color, and font family

Test 2:
1. Test an older version of WooPayments that doesn't have `getAppearance` for regression
1. As an existing WooPay shopper, go to the blocks checkout page and initialize WooPay with the **email field redirect**
2. Should see WooPay support background color, color, and font family
3. Test `Return to cart` -> Blocks checkout page -> WooPay button
4. Should see WooPay support background color, color, and font family
5. Test `Checkout as guest` and initialize WooPay as a new WooPay shopper with the **email field redirect**
6. Should see WooPay support background color, color, and font family
7. Test `Return to cart` -> Blocks checkout page
8. As a new WooPay shopper, initialize WooPay with the WooPay button
9. Should see WooPay support background color, color, and font family

*

-------------------


**Post merge**
